prov/mrail/src/mrail.h

```diff
@@ -192,6 +196,12 @@ struct mrail_ep {
 	struct mrail_recv_fs	*recv_fs;
 	struct mrail_recv_queue recv_queue;
 	struct mrail_recv_queue trecv_queue;
+
+	struct util_buf_pool	*req_pool; 
+	struct slist		deferred_reqs;
+
+	ofi_fastlock_acquire_t	lock_acquire;
+	ofi_fastlock_release_t	lock_release;
```

Comment:
I wonder if these shouldn't move into util_ep, since that's where the lock is defined.